### DISCUSSION TITLE
ShipXplorer AIS Dongle

### DISCUSSION THREAD

**Discussion by aviljoenn** [bareboat-necessities/lysmarine_gen#265] [Q&A] [answered]
I'm at section 1.11 of the OS Documentation which refers to dAISy AIS receiver.
In my case I have a ShipXplorer AIS Dongle.
I do not see after connecting it, I see only my GPS dongle.
Any advice will be appreciated.
See output:
user@lysmarine:~ $ ls -l /dev/ttyLYS*
lrwxrwxrwx 1 root root 7 Dec 27 21:17 /dev/ttyLYS_gps_0 -> ttyACM0
user@lysmarine:~ $

**Comment by mgrouch** [comment]
What is the output of 

`lsusb`

it would be helpful to see productId/vendorId for your device. 
If it uniquely identifies it then it would be possible to construct udev rule for it to create an alias which other programs could use.

**Reply by mgrouch** [reply]
Also 

`AIS-catcher -l`

output could be helpful

**Reply by aviljoenn** [reply]
user@lysmarine:~ $ AIS-catcher -l
AIS-catcher (build Nov  9 2022) v0.40
(C) Copyright 2021-2022 jvde-github and other contributors
This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Found 1 device(s):
0: Realtek, RTL2838UHIDIR, SN: 00000001
user@lysmarine:~ $ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 005: ID 046d:c52b Logitech, Inc. Unifying Receiver
Bus 001 Device 004: ID 1546:01a7 U-Blox AG [u-blox 7]
Bus 001 Device 003: ID 0bda:2838 Realtek Semiconductor Corp. RTL2838 DVB-T
Bus 001 Device 002: ID 2109:3431 VIA Labs, Inc. Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
user@lysmarine:~ $

**Reply by aviljoenn** [reply]
The ShipXplorer seems to be:
Bus 001 Device 003: ID 0bda:2838 Realtek Semiconductor Corp. RTL2838 DVB-T

**Reply by mgrouch** [reply]
From reading about it it seems it’s supposed to spit out AIS NMEA messages on USB. 

can you find 

/dev/tty* 

which corresponds to your AIS stick?

then add it as a NMEA 0183 connection into SignalK with baud rate of 38400

**Reply by aviljoenn** [reply]
In my case, would it be tty3?
ser@lysmarine:~ $ ls /dev/tty*
/dev/tty    /dev/tty11  /dev/tty15  /dev/tty19  /dev/tty22  /dev/tty26  /dev/tty3   /dev/tty33  /dev/tty37  /dev/tty40  /dev/tty44  /dev/tty48  /dev/tty51  /dev/tty55  /dev/tty59  /dev/tty62  /dev/tty9          /dev/ttyprintk
/dev/tty0   /dev/tty12  /dev/tty16  /dev/tty2   /dev/tty23  /dev/tty27  /dev/tty30  /dev/tty34  /dev/tty38  /dev/tty41  /dev/tty45  /dev/tty49  /dev/tty52  /dev/tty56  /dev/tty6   /dev/tty63  /dev/ttyACM0       /dev/ttyS0
/dev/tty1   /dev/tty13  /dev/tty17  /dev/tty20  /dev/tty24  /dev/tty28  /dev/tty31  /dev/tty35  /dev/tty39  /dev/tty42  /dev/tty46  /dev/tty5   /dev/tty53  /dev/tty57  /dev/tty60  /dev/tty7   /dev/ttyAMA0
/dev/tty10  /dev/tty14  /dev/tty18  /dev/tty21  /dev/tty25  /dev/tty29  /dev/tty32  /dev/tty36  /dev/tty4   /dev/tty43  /dev/tty47  /dev/tty50  /dev/tty54  /dev/tty58  /dev/tty61  /dev/tty8   /dev/ttyLYS_gps_0
user@lysmarine:~ $

**Reply by mgrouch** [reply]
Not /dev/tty3

should be /dev/ttyUSBx

or something 

unplug it. See which /dev/tty* disappears


Plug it back and see which gets added.

**Comment by mgrouch** [comment]
Oh. And there is somewhere in the doc 
You might have disable rtlsdr kernel drivers which Linux loads by default if sees rtl device. 
do not do it right away. Try it only if you do not see AIS output on you serial USB port. 

AIS will not output anything if your antenna doesn’t have any ships on clear line of sight.

**Reply by aviljoenn** [reply]
OK, in my case I'm busy doing my setup at home in Wellington South Africa. I will only go sailing again on Friday! (looking forward :-) )

**Reply by mgrouch** [reply]
If your antenna can’t see ships AIS port will be silent. You need to be closer to marine traffic

**Reply by aviljoenn** [reply]
I don't see a tty* file associated to the ShipXplorer. E.g. I see /dev/ttyLYS_gps_0 for my GPS, but nothing for ShipXplorer.
Do you think the file will only appear once a ship comes in range?
Should I wait until I'm on my boat on Friday and try again?

**Reply by mgrouch** [reply]
No. The file should be there regardless of seeing ships. 
ttyACM0 or ttyS0 do not disappear when you unplug?

**Reply by aviljoenn** [reply]
No, unfortunately nothing changes when I plug or unplug the AIS.
However, it seems that both ttyACM0 and ttyLYS_gps_0 comes and goes when I plug and unplug the GPS

**Reply by mgrouch** [reply]
so ttyACM0 is linked as gps_0 via symbolic link. 
No device file for AIS is weird. 

Try disabling drivers as in 
https://bareboat-necessities.github.io/my-bareboat/bareboat-os.html#_rtl_sdr

and reboot

**Reply by mgrouch** [reply]
And if that doesn’t work then contact the vendor of ups shipxplorer stick. You should
refer them to this thread so they would see what we already tried so far.

**Reply by aviljoenn** [reply]
I tried disabling the drivers, but no change yet.
I will reach out to the vendor. 
Thanks so much so far!

**Comment by mgrouch** [comment, marked as answer]
This Amazon review explains how it can be set up. That stick is really not an AIS receiver. 
It’s just SDR which needs software to decode. 

https://p-yo-www-amazon-com-kalias.amazon.com/gp/aw/review/B09Y61CM12/RJG39L0WEYFLY?ref=pf_vv_at_pdctrvw_srp&fbclid=IwAR0UfT42BEreeHd-ULrjAhvn5256cymXqFqZB7vOFmlgogNBRv5OWyy_tdQ&mibextid=ykz3hl
